java/data/PairwiseFilesMerger.java

```diff
@@ -0,0 +1,100 @@
+package data;
+
+import java.io.*;
+import java.nio.file.*;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import static data.Constants.*;
+
+public class PairwiseFilesMerger implements SortedFilesMerger {
+
+    @Override
+    public boolean mergeFiles(List<String> filenames, String inputFile) {
+        try {
+            mergeFilesInPairs(filenames, inputFile);
+        } catch (IOException e) {
+            e.printStackTrace();
+            return false;
+        }
+        try {
+            Files.walk(Paths.get("resources/tmp/"))
+                    .sorted(Comparator.reverseOrder())
```

Comment:
Java constrain in deleting folders that contain files, not optimal since its O(nlogn).
Replaced with 2 deletes, in any case I dont create sub-directories.